pkg/sensors/exec/exec_test.go

```diff
@@ -449,3 +450,53 @@ func TestLoadInitialSensor(t *testing.T) {
 
 	sensors.UnloadAll(tus.Conf().TetragonLib)
 }
+
+func TestDocker(t *testing.T) {
+	if err := exec.Command("docker", "version").Run(); err != nil {
+		t.Skipf("docker not available. skipping test: %s", err)
+	}
+
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	obs, err := observer.GetDefaultObserver(t, ctx, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserver error: %s", err)
+	}
+
+	observer.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+
+	readyWG.Wait()
+	serverDockerID := observer.DockerRun(t, "--name", "fgs-test-server", "--entrypoint", "nc", "quay.io/cilium/alpine-curl:1.0", "-nvlp", "8081", "-s", "0.0.0.0")
+	time.Sleep(1 * time.Second)
+
+	// Tetragon sends 31 bytes + \0 to user-space. Since it might have an arbitrary prefix,
+	// match only on the first 24 bytes.
+	fgsServerID := sm.Prefix(serverDockerID[:24])
+
+	selfChecker := ec.NewProcessChecker().
+		WithBinary(sm.Suffix(tus.Conf().SelfBinary))
+
+	ncSrvChecker := ec.NewProcessChecker().
+		WithBinary(sm.Suffix("/nc")).
+		WithArguments(sm.Full("-nvlp 8081 -s 0.0.0.0")).
+		WithCwd(sm.Full("/")).
+		WithUid(0).
+		WithDocker(fgsServerID)
+
+	checker := ec.NewUnorderedEventChecker(
+		ec.NewProcessExecChecker().
+			WithProcess(selfChecker).
+			WithParent(ec.NewProcessChecker()),
+		ec.NewProcessExecChecker().
+			WithProcess(ncSrvChecker),
+	)
+
+	time.Sleep(1 * time.Second)
```

Comment:
Why do we need this `Sleep()`? (maybe the same question also for the `Sleep()` in line 474)

Reply:
This one isn't needed, I'll remove it. The one on 474 is just to give the process chance to start up and be logged before we check the logs.